plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -0,0 +1,298 @@
+    GridData javaPackagePosition = new GridData(GridData.FILL_HORIZONTAL);
+    javaPackagePosition.horizontalSpan = 2;
+    javaPackageField.setLayoutData(javaPackagePosition);
+    javaPackageField.addModifyListener(pageValidator);
+    
+    // App Engine Project ID
+    Label projectIdLabel = new Label(container, SWT.NONE);
+    projectIdLabel.setText("App Engine Project ID: (optional)"); //$NON-NLS-1$
+    projectIdField = new Text(container, SWT.BORDER);
+    GridData projectIdPosition = new GridData(GridData.FILL_HORIZONTAL);
+    projectIdPosition.horizontalSpan = 2;
+    projectIdField.setLayoutData(projectIdPosition);
+    projectIdField.addModifyListener(pageValidator);
+    
+    setControl(container);
+  }
+
+  protected void openLocationDialog() {
+    DirectoryDialog dialog = new DirectoryDialog(getShell());
+    dialog.setText("Please select the location to contain generated project");
+    String loc = dialog.open();
+    if (loc != null) {
+      locationField.setText(loc);
+      checkPageComplete();
+    }
+  }
+
+  protected void checkPageComplete() {
+    setPageComplete(validatePage());
+  }
+
+  /**
+   * Validate and report on the contents of this page
+   * 
+   * @return true if valid, false if there is a problem
+   */
+  public boolean validatePage() {
+    setMessage(null);
+    setErrorMessage(null);
+    
+    String location = locationField.getText().trim();
+    if (!useDefaults() && location.isEmpty()) {
+      setMessage("Please provide a location", INFORMATION);
+      return false;
+    }
+
+    if (!validateMavenSettings()) {
+      return false;
+    }
+
+    String artifactId = getArtifactId();
+    if (!artifactId.isEmpty()) {
+      IPath path = useDefaults() ? getWorkspace().getRoot().getLocation() : new Path(location);
+      path = path.append(artifactId);
+      // don't overwrite a location
+      if (path.toFile().exists()) {
+        setErrorMessage(MessageFormat.format("Location already exists: {0}", path));
+        return false;
+      }
+    }
+
+    String packageName = getPackageName();
+    if (!JavaPackageValidator.validate(packageName)) {
+      setErrorMessage("Illegal Java package name: " + packageName); //$NON-NLS-1$
+      return false;
+    }
+
+    String projectId = getAppEngineProjectId();
+    if (!AppEngineProjectIdValidator.validate(projectId)) {
+      setErrorMessage("Illegal App Engine Project ID: " + projectId); //$NON-NLS-1$
+      return false;
+    }
+
+    return true;
+  }
+
+  private boolean validateMavenSettings() {
+    String groupId = getGroupId();
+    if (groupId.isEmpty()) {
+      setMessage("Please provide Maven group id", INFORMATION);
+      return false;
+    } else if (!MavenCoordinatesValidator.validateGroupId(groupId)) {
+      setErrorMessage("Illegal group id: " + groupId);
+      return false;
+    }
+    String artifactId = getArtifactId();
+    if (artifactId.isEmpty()) {
+      setMessage("Please provide Maven artifact id", INFORMATION);
```

Comment:
Good point. But I'll follow the Maven style for Maven pieces to match their dialogs.  Will use Project ID for our App Engine identifier though